app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,72 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+
+    if previous_live_version
```

Comment:
Can we do `edition.temporary_first_published_at = previous_live_version&.first_published_at` here to avoid the if?